### ISSUE TITLE
setGroupBy doesn't work correctly in some cases

### ISSUE THREAD

**Issue by Narretz** [olifolkerd/tabulator#3666]
**Describe the bug**
If I use setGroupBy() after init, with either array or string, weird stuff happens. In the example, the rows are duplicated for some reason, but in my actual app nothing happened instead.

**Tabulator Info**
latest from cdn

**Working Example**
https://codepen.io/narretz/pen/ZEvbxya

**To Reproduce**
You should immediately see that the group rows are duplicated, but only one set has actual content.

**Desktop (please complete the following information):**
 - latest firefox

**Comment by bornova** [comment]
@Narretz , you may need to set grouping after the table is built. 
```javascript
table.on('tableBuilt', () => {
  table.setGroupBy('age')
})
```
However above code only works if you have the `groupBy` option initially set in table options (commented out in your codepen).  If this option is missing, calling `setGroupBy` twice seems to work for some reason but I am not sure if this was the intended behaviour or a bug.  Following works:

```javascript
table.on('tableBuilt', () => {
  table.setGroupBy(true)
  table.setGroupBy('age')
})
```

**Comment by Narretz** [comment]
Ah, so it's a good old race condition.

I didn't know about `table.on('tableBuilt')`. Maybe there should be a warning when you use commands before the table is built? Or is this supposed to work?

> However above code only works if you have the groupBy option initially set in table options (commented out in your codepen). 

That's actually what I saw in my actual app, the setGroup was behind a button click. I fixed that by setting groubBy to [] on init.

**Comment by tadmi** [comment]
same problem when use setGroupBy() after table init

**Comment by olifolkerd** [comment]
This will occur with any function called on the table before the table built event, except the 'on' function.

Please don't list them all here it will literally be all of them.

This is mentioned in several places in the docs already but I will make it more prominent

**Comment by Narretz** [comment]
Wouldn't printing a warning or letting the command fail make this most obvious?

**Comment by olifolkerd** [comment]
It currently does on the core Tabulator commands. Module added functions
will be getting the warning in the upcoming 5.2 release.

Cheers

Oli

**Comment by olifolkerd** [comment]
I have pushed an update to the 5.2 branch that makes the initialization warning messages more consistent across all functions.

It will be included in the 5.2 release later this month.

Cheers

Oli :)
